### PR TITLE
Refactor: use PIL as default image format

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

follow-up to #1313 as the Tensor formating is not very friendly to other general augmentation packages

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [x] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/data/utils.py [hunk 1]
```diff
@@ -162,7 +164,7 @@ def image_default_loader(file_path: str, drop_alpha: bool = True) -> Image:
         file_path: The image file to load.
         drop_alpha: If ``True`` (default) then any alpha channels will be silently removed.
     """
-    img = default_loader(file_path)
+    img = default_loader(file_path) if _TORCHVISION_AVAILABLE else pil_image.open(file_path)
```
Comment (krshrimali) [inline flash/core/data/utils.py:167]: From my suggestion above:

```suggestion
    img = default_loader(file_path) if _TORCHVISION_AVAILABLE else PILImage.open(file_path)
```

**Review by krshrimali — COMMENTED** [review]
Part-1 of my review. I'll appreciate it if you can briefly explain why this was required, and what was the problem before (it will be easier to review and land if all looks good). :)

File: flash/core/data/transforms.py [hunk 2]
```diff
@@ -37,8 +42,10 @@ def __init__(self, keys: Union[str, Sequence[str]], *args):
         self.keys = keys
 
     def forward(self, x: Mapping[str, Any]) -> Mapping[str, Any]:
+        if not isinstance(x[DataKeys.INPUT], Tensor):
+            x[DataKeys.INPUT] = to_tensor(x[DataKeys.INPUT])
```
Comment (ethanwharris) [inline flash/core/data/transforms.py:46]: Definitely shouldn't do this. INPUT may not always be an image (could be text for example). We should just manually include a `ToTensor` transform in the input transforms that need it

**Review by ethanwharris — COMMENTED** [review]
Awesome! Few comments. Let's merge this before updating #1313 so that it can work directly with the PIL image

File: flash/core/data/utils.py [hunk 3]
```diff
@@ -70,8 +73,7 @@ def download_data(url: str, path: str = "data/", verbose: bool = False) -> None:
     # __author__  = "github.com/ruxi"
     # __license__ = "MIT"
 
-    Examples
-    ________
+    Examples:
```
Comment (ethanwharris) [inline flash/core/data/utils.py:76]: We tend to use the underline everywhere. Any particular reason for this change?

Comment (Borda) [reply flash/core/data/utils.py:76]: not really... sorry :)

File: flash/core/data/transforms.py (hunk 2 above)
Comment (Borda) [reply flash/core/data/transforms.py:46]: note that this is a deprecated approach and kept only for legacy reasons...

**Comment by krshrimali** [comment]
Hi, @Borda - Would you mind adding tests to this PR? Just a sample test will work, even if it fails, I'll take it from there. Thanks!

**Comment by krshrimali** [comment]
> Hi, @Borda - Would you mind adding tests to this PR? Just a sample test will work, even if it fails, I'll take it from there. Thanks!

Hi, @Borda - Just a gentle ping, if you can write a test (as I mentioned before - no need to ensure that it passes, just a template test that helps me understand what you are trying to achieve here will help) - that will be great! Please ping me once you are done, also don't worry about the merge conflicts. I'll take care of it, just force push the test. :)

**Comment by Borda** [comment]
@krshrimali @ethanwharris may pls get your help on this collision resolution... :rabbit:

**Review by ethanwharris — APPROVED** [review]

**Pull request merged**